src/OAuth/CredentialStorage/InMemoryCredentialStorage.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license.
+ */
+
+namespace ZfrLightspeedRetail\OAuth\CredentialStorage;
+
+use ZfrLightspeedRetail\OAuth\Credential;
+
+/**
+ * @author Daniel Gimenes
+ */
+final class InMemoryCredentialStorage implements CredentialStorageInterface
```

Comment:
How an InMemoryCredentialStorage could be useful? Any other request coming after will loose the refresh token and will be unable to authenticate.

Unfortunately I don't see any other credential method than storing into a persistence layer (even APCu would not work due to the lack of shared processes, and Memcached neither due to the possible eviction of cache keys).

I think we should even have a default "DynamoDbCredentialStorage" (as it's quite perfect for that kind of use cases). Just make it configurable to allow to personalize the table name (by defaulting it to something like "lightspeed-tokens". And add the aws-sdk as a dev dependencies and a suggest.